soh/soh/OTRGlobals.cpp

```diff
@@ -925,6 +927,84 @@ bool PathTestCleanup(FILE* tfile) {
     return true;
 }
 
+// TODO: Copied from randomzier, find a better place for it
+std::string sanitizePath(std::string stringValue) {
+    // Add backslashes.
+    for (auto i = stringValue.begin();;) {
+        auto const pos = std::find_if(i, stringValue.end(), [](char const c) { return '\\' == c || '\'' == c || '"' == c; });
+        if (pos == stringValue.end()) {
+            break;
+        }
+        i = std::next(stringValue.insert(pos, '\\'), 2);
+    }
+
+    // Removes others.
+    stringValue.erase(std::remove_if(stringValue.begin(), stringValue.end(), [](char const c) {
+        return '\n' == c || '\r' == c || '\0' == c || '\x1A' == c; }), stringValue.end());
+
+    return stringValue;
+}
+
+static const char* lastDroppedFile = "";
+
+void processDroppedFiles() {
+    const char* droppedFile = CVarGetString("gDroppedFile", "");
+    if (strcmp(droppedFile, lastDroppedFile) != 0 && strcmp(droppedFile, "") != 0) {
+        lastDroppedFile = droppedFile;
+        try {
+            std::ifstream configStream(sanitizePath(droppedFile));
+            if (!configStream) {
+                return;
+            }
+
+            nlohmann::json configJson;
+            configStream >> configJson;
+
+            if (!configJson.contains("CVars")) {
+                return;
+            }
+
+            clearCvars(enhancementsCvars);
+            clearCvars(cheatCvars);
+            clearCvars(randomizerCvars);
+
+            // Flatten everything under CVars into a single array
+            auto cvars = configJson["CVars"].flatten();
+
+            for (auto& [key, value] : cvars.items()) {
+                // Replace slashes with dots in key, and remove leading dot
+                std::string path = key;
+                std::replace(path.begin(), path.end(), '/', '.');
+                if (path[0] == '.') {
+                    path.erase(0, 1);
+                }
+                if (value.is_string()) {
+                    CVarSetString(path.c_str(), value.get<std::string>().c_str());
+                } else if (value.is_number_integer()) {
+                    CVarSetInteger(path.c_str(), value.get<int>());
+                } else if (value.is_number_float()) {
+                    CVarSetFloat(path.c_str(), value.get<float>());
+                }
+            }
+
+            auto gui = LUS::Context::GetInstance()->GetWindow()->GetGui();
+            gui->GetGuiWindow("Console")->Hide();
+            gui->GetGuiWindow("Actor Viewer")->Hide();
+            gui->GetGuiWindow("Collision Viewer")->Hide();
+            gui->GetGuiWindow("Save Editor")->Hide();
+            gui->GetGuiWindow("Display List Viewer")->Hide();
+            std::reinterpret_pointer_cast<LUS::ConsoleWindow>(LUS::Context::GetInstance()->GetWindow()->GetGui()->GetGuiWindow("Console"))->ClearBindings();
```

Comment:
should probably use a dynamic instead of a reinterpret here

<https://stackoverflow.com/questions/332030/when-should-static-cast-dynamic-cast-const-cast-and-reinterpret-cast-be-used>

> dynamic_cast is exclusively used for handling polymorphism

`ConsoleWindow` is a `GuiWindow` so that fits

> reinterpret_cast is the most dangerous cast, and should be used very sparingly.

Reply:
Could you show how this can be used? I'm having trouble (just copied this from other places console window is used)

Reply:
https://github.com/Kenix3/libultraship/blob/1b41afa9eaedb19a9073f5dd1eca2cda4dea59c8/src/controller/deviceindex/LUSDeviceIndexMappingManager.cpp#L547-L550

Reply:
seems LUS must have a `using namespace std` hiding somewhere, here's an example with std:: in it https://github.com/HarbourMasters/Shipwright/blob/d0d1d9c4879c4f95622a9573b171a6bf13f55597/soh/soh/Enhancements/controls/SohInputEditorWindow.cpp#L294